source/CustomSale.cpp

```diff
@@ -0,0 +1,287 @@
+}
+
+
+
+void CustomSale::Load(const DataNode &node, const Set<Sale<Outfit>> &items, const Set<Outfit> &outfits)
+{
+	for(const DataNode &child : node)
+	{
+		const std::string &token = child.Token(0);
+		if((token == "clear" || token == "remove"))
+		{
+			if(child.Size() == 1)
+				clear();
+			else if(child.Token(1) == "outfit" && child.Size() >= 3)
+			{
+				const Outfit *outfit = outfits.Get(child.Token(2));
+				relativeOutfitPrices.erase(outfit);
+				relativeOutfitOffsets.erase(outfit);
+			}
+			else if(child.Token(1) == "outfitter" && child.Size() >= 3)
+			{
+				const Sale<Outfit> *item = items.Get(child.Token(2));
+				relativePrices.erase(item);
+				relativeOffsets.erase(item);
+			}
+		}
+		else if(token == "outfit")
+		{
+			for(const DataNode &grandChild : child)
+			{
+				bool value = (grandChild.Token(0) == "value");
+				bool offset = (grandChild.Token(0) == "offset");
+				if(value || offset)
+					for(const DataNode &kid : grandChild)
+					{
+						bool add = (kid.Token(0) == "add");
+						const Outfit *outfit = outfits.Get(kid.Token(add));
+						
+						if(kid.Size() < 2 + add)
+							continue;
+						if(value)
+						{
+							if(add)
+								relativeOutfitPrices[outfit] += kid.Value(2);
+							else
+								relativeOutfitPrices[outfit] = kid.Value(1);
+							if(child.Size() == 2 + add)
+								relativeOutfitPrices[outfit] /= outfit->Cost();
+						}
+						else if(offset)
+						{
+							if(add)
+								relativeOutfitOffsets[outfit] += kid.Value(2);
+							else
+								relativeOutfitOffsets[outfit] = kid.Value(1);
+							if(child.Size() == 2 + add)
+								relativeOutfitOffsets[outfit] /= outfit->Cost();
+						}
+					}
+				else if(grandChild.Size() >= 2)
+				{
+					const Outfit *outfit = outfits.Get(grandChild.Token(0));
+					relativeOutfitPrices[outfit] = grandChild.Value(1) / outfit->Cost();
+				}
+			}
+		}
+		else if(token == "outfitter")
+		{
+			for(const DataNode &grandChild : child)
+			{
+				bool value = (grandChild.Token(0) == "value");
+				bool offset = (grandChild.Token(0) == "offset");
+				if(value || offset)
+					for(const DataNode &kid : grandChild)
+					{
+						bool add = (kid.Token(0) == "add");
+						const Sale<Outfit> *item = items.Get(kid.Token(add));
+
+						if(kid.Size() < 2 + add)
+							continue;
+						if(value)
+						{
+							if(add)
+								relativePrices[item] += kid.Value(2);
+							else
+								relativePrices[item] = kid.Value(1);
+						}
+						else if(offset)
+						{
+							if(add)
+								relativeOffsets[item] += kid.Value(2);
+							else
+								relativeOffsets[item] = kid.Value(1);
+						}
+					}
+				else if(grandChild.Size() >= 2)
+				{
+					const Sale<Outfit> *item = items.Get(grandChild.Token(0));
+					relativePrices[item] = grandChild.Value(1);
+				}
+			}
+		}
+		else if(token == "hidden" || token == "import")
+		{
+			for(const auto& it : show)
+				if(token == it.second)
+					sellType = it.first;
+		}
```

Comment:
I will change it to be that way, I thought it would be better in a more generic way so that later on if we want to add a new keyword it would be easy to just add `token == "new token"` within the thing but clarity is more important here as you say, only 3 different types exist after all.